# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteVideoDecoder.cpp
# Commit: 880f0718c9ba
# Full Hash: 880f0718c9bac68a2649620ce9621fc06cc40b25
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2022-09-12 21:40:55
# Regressor Bug: 1787561
# File Overlap Count: 1
# Description:
#   Bug 1790281: Fixup a bad merge from Bug 1787561. r=alwu
#   
#   When D155902 merged and landed, it undid a recent change in
#   RemoteVideoDecoderParent::ProcessDecodedData without causing a compiler
#   error. This fixes up the merge to what was intended.
# ==============================================================================

diff -r 8de3f54dc3ca -r 880f0718c9ba dom/media/ipc/RemoteVideoDecoder.cpp
--- a/dom/media/ipc/RemoteVideoDecoder.cpp	Mon Sep 12 17:22:24 2022 +0000
+++ b/dom/media/ipc/RemoteVideoDecoder.cpp	Mon Sep 12 17:24:48 2022 +0000
@@ -274,10 +274,14 @@
         MediaDataIPDL(data->mOffset, data->mTime, data->mTimecode,
                       data->mDuration, data->mKeyframe),
         video->mDisplay,
-        RemoteImageHolder(mParent,
-                          XRE_IsGPUProcess() ? VideoBridgeSource::GpuProcess
-                                             : VideoBridgeSource::RddProcess,
-                          size, video->mImage->GetColorDepth(), sd),
+        RemoteImageHolder(
+            mParent,
+            XRE_IsGPUProcess()
+                ? VideoBridgeSource::GpuProcess
+                : (XRE_IsRDDProcess()
+                       ? VideoBridgeSource::RddProcess
+                       : VideoBridgeSource::MFMediaEngineCDMProcess),
+            size, video->mImage->GetColorDepth(), sd),
         video->mFrameID);
 
     array.AppendElement(std::move(output));
